torch_object_detection_segmenter.py

```diff
@@ -1,31 +1,209 @@
 __copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
-from typing import Optional
+
+from jina.logging.logger import JinaLogger
 import numpy as np
+from typing import Dict, List, Union, Tuple, Optional, Any
+import torchvision.models.detection as detection_models
+
+from jina import Document, DocumentArray, Executor, requests
 
-from jina import Executor, DocumentArray, requests
 
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i: i + batch_size]
 
 class TorchObjectDetectionSegmenter(Executor):
     """
-    Encode the Document blob into embedding.
-
-    :param embedding_dim: the output dimensionality of the embedding
+    :class:`TorchObjectDetectionSegmenter` detects objects
+    from an image using `torchvision detection models`
+    and crops the images according tothe detected bounding boxes
+    of the objects with a confidence higher than a threshold.
+    :param model_name: the name of the model. Supported models include
+        ``fasterrcnn_resnet50_fpn``, ``maskrcnn_resnet50_fpn`
+    :param channel_axis: the axis id of the color channel,
```

Comment:
I would like a clarification. This indicates at which axis the image should come? I think this should be out of the executor

Reply:
clarified in README